test/box-luatest/gh_8192_feedback_daemon_metrics_test.lua

```diff
@@ -0,0 +1,141 @@
+local t = require('luatest')
+local g = t.group('Feedback daemon metrics')
+local server = require('luatest.server')
+
+g.before_all(function()
+    g.server = server:new{alias = 'default'}
+    g.server:start()
+end)
+
+g.after_all(function()
+    g.server:drop()
+end)
+
+g.before_each(function()
+    g.server:exec(function()
+        package.loaded["metrics"] = nil
+        -- Daemon is reloaded when cfg option is updated.
+        box.cfg{
+            feedback_send_metrics = true,
+            feedback_metrics_collect_interval = 0.01,
+            feedback_metrics_limit = 1024 * 1024
+        }
+    end)
+end)
+
+g.test_metrics = function()
+    g.server:exec(function()
+        local t = require('luatest')
+        local fiber = require('fiber')
+        local m = {}
+        -- Method checks that we pass required arguments to collect defaults.
+        -- Note that if one of assertions in the method fails, you will not
+        -- see it directly - the method is called in internal fiber. Metrics
+        -- just will not be collected and metrics checkers will fail.
+        m.collect = function(args)
+            t.assert_type(args, 'table')
+            t.assert_equals(args.invoke_callbacks, true)
+            t.assert_equals(args.default_only, true)
+            return {1, 2}
+        end
+        m._VERSION = "0.16.0"
+        package.loaded["metrics"] = m
+        fiber.sleep(0.3)
+        local daemon = box.internal.feedback_daemon
+        local fb = daemon.generate_feedback()
+        -- This check may fail because of assertion failure in collect method.
+        t.assert_type(fb.metrics, 'table')
+        t.assert_almost_equals(#fb.metrics, 30, 5)
+        fb = daemon.generate_feedback()
+        -- Check is metrics were reset. Fiber yields on feedback generation, so
+        -- metrics may be not empty.
+        t.assert(fb.metrics == nil or #fb.metrics < 5)
```

Comment:
Why don't you set a big enough `feedback_metrics_collect_interval` to test both cases?

Reply:
Feedback daemon is cleared on reconfiguring, so I will need to wait for a long time before the first collection.1

Reply:
I don't get your point, could you please elaborate? AFAIU, metrics are collected every `feedback_metrics_collect_interval`, and you explicitly retrieve them by calling `daemon.generate_feedback` — so why will you have to wait long?

Reply:
Because I wait until metrics are collected, and only then I check that they are empty (or almost empty) - the point of this test case is to check if metrics are reset after retrieving.